### PR TITLE
Added settings for queue peeker delay

### LINKED ISSUE #271: add code API for controlling QueuePeeker peek interval
https://github.com/Particular/NServiceBus.SqlServer/blob/3.0.0-Beta2/src/NServiceBus.SqlServer/Receiving/QueuePeeker.cs#L47

propose supported range: 100ms to 10s

### PR DESCRIPTION
**Pull request by HEskandari**
Resolves #271

### CODE REVIEW

**Comment by HEskandari** [comment]
@Particular/sqlserver-transport-maintainers Please review.

File: src/NServiceBus.SqlServer/Receiving/QueuePeekerOptions.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            }
+
+            Delay = delay;
+        }
+
+        static void Validate(TimeSpan delay)
+        {
+            if (delay < TimeSpan.FromMilliseconds(100))
+            {
+                var message = "Delay requested is invalid. The value should be greater than 100 ms and less than 10 seconds.";
```
Comment (tmasternak) [inline src/NServiceBus.SqlServer/Receiving/QueuePeekerOptions.cs]: Can we have a single message for both cases?

File: src/NServiceBus.SqlServer/Receiving/QueuePeeker.cs [hunk 2]
```diff
@@ -42,9 +43,9 @@ public QueuePeeker(SqlConnectionFactory connectionFactory)
             return messageCount;
         }
 
-        SqlConnectionFactory connectionFactory;
+        readonly SqlConnectionFactory connectionFactory;
```
Comment (SzymonPobiega) [inline src/NServiceBus.SqlServer/Receiving/QueuePeeker.cs]: redundant `readonly`

**Comment by tmasternak** [comment]
@HEskandari I think we need a doco for the new method as well

**Comment by tmasternak** [comment]
We should be passing the setting also to `LegacyQueuePeeker`

**Comment by HEskandari** [comment]
@tmasternak oh forgot that one completely :) I'll update.

**Comment by tmasternak** [comment]
The current approach will set the same peek interval for all queues which I think is good. (just confirming ;))

File: src/NServiceBus.SqlServer/Receiving/QueuePeekerOptions.cs (hunk 1 above)
Comment (HEskandari) [inline src/NServiceBus.SqlServer/Receiving/QueuePeekerOptions.cs]: @tmasternak wouldn't that be confusing if the message says the value is "invalid" but still runs? That was why I have used a slightly different wording.

Comment (tmasternak) [inline src/NServiceBus.SqlServer/Receiving/QueuePeekerOptions.cs]: That makes sense @HEskandari. Thanks for clarification.

**Pull request closed without merging**